### PR TITLE
Helpx_Viewport Specific Container cases

### PR DESCRIPTION
**Pull request by kirupacommit**
vps

### CODE REVIEW

**Comment by kirupacommit** [comment]
@skumar09 @amauch-adobe  Kindly review above PR

File: tests/helpx/vps.test.js
Comment (amauch-adobe) [inline tests/helpx/vps.test.js:1]: Just two comments, throughout the test file there was viewport config changes (which is totally fine), but I wanted to point out you can do the same thing for a project config and name it specific to the type of viewport size that will run all tests in that specific viewport.  That being said you may have only wanted this test to be run this way with this viewport without having to send in specific tags for execution to use the project config and filter it down to just this test.

Second comment, is always be careful about the usage of waitForTimeout as stated in playwrights documentation here (https://playwright.dev/docs/api/class-page#page-wait-for-timeout), it is a general recommendation to not use it in production or to waitFor a specific event or expected behavior with an element, or to use locators to happen before running the assertion expectation.  Timeout waits may cause false failures/flakiness due to VM starts, network situations, etc. in CICD pipelines.

**Review by amauch-adobe — APPROVED** [review]
Approve.

File: selectors/helpx/vps.page.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+export default class vps {
+  constructor(page) {
+    this.page = page;
+
+    // vps Selectors for hidden desktop
+    this.vpsForHiddenDesktop = page.locator("//div[contains(@daa-lh,'b2|generic')][contains(@class,'generic hidden-desktop')]");
```
Comment (Dli3) [inline selectors/helpx/vps.page.js:6]: You could chain the locators.
https://playwright.dev/docs/locators#matching-inside-a-locator   
For example: 
`this.vpsForImginHiddenDesktop = this. vpsForHiddenDesktop .locator("/descendant::picture/img")`

File: tests/helpx/vps.test.js [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+  //create a new page
+  page = await context.newPage();
+  vps = new vpsPage(page); 
+  
+});
+
+test.describe('Verify hidden Tablet and mobile and Desktop for video in Viewport specific container', () => {
+
+  test(`${features[7].name}, ${features[7].tags}`, async ({ baseURL }) => {
+    let vpsTag = features[7].path;
```
Comment (Dli3) [inline tests/helpx/vps.test.js:35]: vpsTag doesn't seem to be changing within your test. 
```suggestion
    const vpsTag = features[7].path;
```

File: tests/helpx/vps.test.js [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+          await expect(vps.linkForHiddenTabletMobileB26).not.toBeVisible();
+
+          // Wait for 1000 milliseconds
+          await page.waitForTimeout(1000);
+
+          // Set viewport size for tablet
+          await page.setViewportSize({ width: 768, height: 1024 }); // iPad viewport size
+          await expect(vps.vpsForHiddenTabletMobileB26).not.toBeVisible();
+          await expect(vps.linkForHiddenTabletMobileB26).not.toBeVisible();
+
```
Comment (Dli3) [inline tests/helpx/vps.test.js:62]: Do we need these empty lines?

File: tests/helpx/vps.test.js [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+test.describe('Verify hidden desktop in Viewport specific container', () => {
+
+  test(`${features[0].name}, ${features[0].tags}`, async ({ baseURL }) => {
+    let vpsTag = features[0].path;
+    await page.goto(`${helpxconfig.use?.baseURL}${vpsTag}`);
+    await page.waitForLoadState('domcontentloaded');
+    console.log(`[Test Page]: ${baseURL}${vpsTag}`);
+    //I verify url of the page
+    await test.step('Navigate to Viewport specific container page', async () => {
+      await expect(page).toHaveURL(`${baseURL}${vpsTag}`);
```
Comment (Dli3) [inline tests/helpx/vps.test.js:77]: Do you need the `${helpxconfig.use?.`part that you have above?

File: tests/helpx/vps.test.js [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+});
+
+test.describe('Verify hidden desktop in Viewport specific container', () => {
+
+  test(`${features[0].name}, ${features[0].tags}`, async ({ baseURL }) => {
+    let vpsTag = features[0].path;
+    await page.goto(`${helpxconfig.use?.baseURL}${vpsTag}`);
+    await page.waitForLoadState('domcontentloaded');
+    console.log(`[Test Page]: ${baseURL}${vpsTag}`);
+    //I verify url of the page
```
Comment (Dli3) [inline tests/helpx/vps.test.js:75]: Could be a part of your test.step name.

File: tests/helpx/vps.test.js [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+
+    await page.setViewportSize({ width: 1280, height: 800 }); // Typical desktop viewport size
+
+    //Hidden Desktop
+    await expect(vps.vpsForHiddenTabletB4).toBeVisible();
+    await expect(vps.vpsForImginHiddenTabletB4).toBeVisible();
+
+       await page.waitForTimeout(1000);
+      // Set viewport size for mobile
+      await page.setViewportSize({ width: 375, height: 812 }); // iPhone X viewport size
```
Comment (Dli3) [inline tests/helpx/vps.test.js:125]: You have two very similar comments. Maybe remove one of them?

File: tests/helpx/vps.test.js (hunk 3 above)
Comment (kirupacommit) [reply tests/helpx/vps.test.js:62]: Removed

File: tests/helpx/vps.test.js (hunk 6 above)
Comment (kirupacommit) [reply tests/helpx/vps.test.js:125]: Done. Thanks For your suggestion

File: tests/helpx/vps.test.js (hunk 4 above)
Comment (kirupacommit) [reply tests/helpx/vps.test.js:77]: Thank you for your insight, removed ${helpxconfig.use?}

**Comment by kirupacommit** [comment]
@skumar09 can u please review my PR with updated changes

**Pull request merged**